D4Companion.Services/ScreenProcessHandler.cs

```diff
@@ -130,867 +113,514 @@ private void initImageList()
+        [LogTime]
+        private bool FindTooltips(Bitmap currentScreen)
+        {
+            var currentScreenImg = currentScreen.ToImage<Bgr, byte>();
+            var currentScreenGray = currentScreenImg.Convert<Gray, byte>();
+            var itemTooltipBag = new ConcurrentBag<ItemTooltipDescriptor>();
 
-            ConcurrentBag<ItemTooltipDescriptor> itemTooltipBag = new ConcurrentBag<ItemTooltipDescriptor>();
             Parallel.ForEach(_imageListItemTooltips.Keys, itemTooltip =>
             {
-                try
-                {
-                    itemTooltipBag.Add(FindTooltip(currentScreenFilter, itemTooltip));
-                }
-                catch (Exception exception)
-                {
-                    _logger.LogError(exception, $"{MethodBase.GetCurrentMethod()?.Name}");
-                }
+                var toolTip = FindTooltip(currentScreenGray, itemTooltip);
+                
+                if (toolTip != null) itemTooltipBag.Add(toolTip);
             });
 
-            // Sort results by similarity
-            var itemTooltips = itemTooltipBag.ToList();
-            itemTooltips.Sort((x, y) =>
-            {
-                return x.Similarity < y.Similarity ? -1 : x.Similarity > y.Similarity ? 1 : 0;
-            });
+            var foundToolTip = default(ItemTooltipDescriptor?);
 
-            foreach (var itemTooltip in itemTooltips)
+            foreach (var itemTooltip in itemTooltipBag)
             {
-                if (itemTooltip.Location.IsEmpty) continue;
+                if (foundToolTip == null || foundToolTip.Similarity < itemTooltip.Similarity)
+                    foundToolTip = itemTooltip;
+            }
 
-                _currentTooltip.Location = itemTooltip.Location;
+            var processedScreen = currentScreen;
 
-                Point[] points = new Point[]
-                {
-                    new Point(itemTooltip.Location.Left,itemTooltip.Location.Bottom),
-                    new Point(itemTooltip.Location.Right,itemTooltip.Location.Bottom),
-                    new Point(itemTooltip.Location.Right,itemTooltip.Location.Top),
-                    new Point(itemTooltip.Location.Left,itemTooltip.Location.Top)
-                };
-                CvInvoke.Polylines(currentScreen, points, true, new MCvScalar(0, 0, 255), 5);
-
-                // Skip foreach after the first valid tooltip is found.
-                break;
+            if (foundToolTip != null)
+            {
+                _currentTooltip.Location = foundToolTip.Location;
+                _currentScreenTooltip = currentScreenImg.Copy(_currentTooltip.Location);
+                _currentScreenTooltipBin = _currentScreenTooltip
+                    .Convert<Gray, byte>()
+                    .ThresholdBinaryInv(_thresholdMin, _thresholdMax);
+
+                DrawRect(currentScreenImg, foundToolTip.Location);
```

Comment:
See notes on line 200/205.
No need for an extra one-line function. The static `CvInvoke.Rectangle` is already OK.